commcare_connect/opportunity/models.py

```diff
@@ -67,13 +67,34 @@ def __str__(self):
 
     @property
     def remaining_budget(self) -> int:
+        return self.total_budget - self.claimed_budget
+
+    @property
+    def claimed_budget(self):
+        return self.claimed_visits * self.budget_per_visit
+
+    @property
+    def utilised_budget(self):
+        return self.approved_visits * self.budget_per_visit
+
+    @property
+    def claimed_visits(self):
         opp_access = OpportunityAccess.objects.filter(opportunity=self)
         used_budget = OpportunityClaim.objects.filter(opportunity_access__in=opp_access).aggregate(
             Sum("max_payments")
         )["max_payments__sum"]
         if used_budget is None:
             used_budget = 0
-        return self.total_budget - used_budget
+        return used_budget
+
+    @property
+    def approved_visits(self):
+        approved_user_visits = (
+            UserVisit.objects.filter(opportunity=self, status=VisitValidationStatus.approved)
```

Comment:
I think this is redundant. It counts only `approved` visits and excludes `over_limit`, but since a visit can't be both, we don't need the `exclude` statement.

Reply:
Removed. https://github.com/dimagi/commcare-connect/pull/149/commits/76ef45bcdcc0d69cfa5bfb567668b70905d92908